### PR TITLE
Create api routes

### PR DESCRIPTION
**Pull request by kemalfidan**
This task wasn't in Jira but I thought it'd help us to have the layout of the api structured. There is some overlapping commits between this PR and the one before. The most important change for this one is that I've defined more endpoints under `pages/api/`.

### CODE REVIEW

**Comment by kemalfidan** [comment]
Also I fixed the errors on my previous PR so I'll merge those updates here once they get approved.

**Review by rossketron — APPROVED** [review]
LGTM 👍

File: server/models/Event.ts
Comment (andlrutt) [inline server/models/Event.ts:8]: Do we need expand this to have it mirror `/utils/types.ts` ?

**Review by andlrutt — APPROVED** [review]
Looks great! The outlines of the endpoints will be very helpful going forward.

**Review by zaviermiller — COMMENTED** [review]
api routes look good.

**Review by zaviermiller — APPROVED** [review]

**Pull request merged**
